libr/core/cmd_anal.c

```diff
@@ -674,7 +674,10 @@ static void type_cmd(RCore *core, const char *input) {
 		r_core_cmd0 (core, "aei");
 		r_core_cmd0 (core, "aeim");
 		r_reg_arena_push (core->anal->reg);
-		r_list_foreach (core->anal->fcns, it, fcn) {
+		RList *fcns = r_list_clone (core->anal->fcns);
+		// Reversing list so that we get function in top-bottom call order
+		r_list_reverse (fcns);
+		r_list_foreach (fcns, it, fcn) {
```

Comment:
Why not just foreach_backwards?

Reply:
Uhmm i didn't know such thing existed, now updated the PR !

but btw u gave merged the r2r pr , before this i guess :D